src/teleop/gui/src/components/Chlorophyll.vue

```diff
@@ -0,0 +1,184 @@
+                    <col style="width: 53px">
+                    <col style="width: 53px">
+                    <col style="width: 53px">
+                    <col style="width: 53px">
+                    <col style="width: 53px">
+                </colgroup>
+                <thead>
+                <tr>
+                    <th class = "tableElement"></th>
+                    <th class = "tableElement">1</th>
+                    <th class = "tableElement">2</th>
+                    <th class = "tableElement">3</th>
+                    <th class = "tableElement">4</th>
+                    <th class = "tableElement">5</th>
+                    <th class = "tableElement">6</th>
+                </tr>
+                </thead>
+                <tbody>
+                    <tr>
+                        <td class = "tableElement">Spec 0</td>
+                        <td class = "tableElement" v-for="i in 6" :key="i">{{ (spectral_data[i-1]).toFixed(0) }}</td>
+                    </tr>
+                </tbody>
+            </table>
+        </div>
+        <GenerateReport v-bind:spectral_data="spectral_data"/>
+    </div>
+</template>
+
+<script>
+import ToggleButton from './ToggleButton.vue';
+import ROSLIB from 'roslib';
+import GenerateReport from './GenerateReport.vue';
+
+export default {
+    data() {
+        return {
+            whiteLEDS_active: false,
+            UV_endEffector: false,
+            UV_carousel: false
+        }
+    },
+
+    props: {
+        spectral_data: {
+            type: Array,
+            required: true
+        }
+    },
+
+    components: {
+        ToggleButton,
+        GenerateReport
+    },
+
+    methods: {
+        toggle_whiteLEDS: function () {
+            this.whiteLEDS_active = !this.whiteLEDS_active
+            let whiteLedService = new ROSLIB.Service({
+                ros : this.$ros,
+                name : 'change_white_led_state',
+                serviceType : 'mrover/ChangeDeviceState'
+            });
+            let request = new ROSLIB.ServiceRequest({
+                enable: this.whiteLEDS_active
+            });
+            whiteLedService.callService(request, (result) => {
+                if (!result) {
+                    this.whiteLEDS_active = !this.whiteLEDS_active
+                    alert("Toggling white LEDs failed.")
+                }
+            });
+        },
+
+        toggleEndEffectorUV: function () {
+            this.UV_endEffector = !this.UV_endEffector
+            let uvService = new ROSLIB.Service({
+                ros : this.$ros,
+                name : 'change_uv_led_end_effector_state',
+                serviceType : 'mrover/ChangeDeviceState'
+            });
+            let request = new ROSLIB.ServiceRequest({
+                enable: this.UV_endEffector
+            });
+            uvService.callService(request, (result) => {
+                if (!result) {
+                    this.UV_endEffector = !this.UV_endEffector
```

Comment:
same with this as stated above. basically, let's say the uv end effector is currently on. the button is currently off. if the user tries to turn on the end effector and the response is "failure" then this code changes it back to off unnecessarily (since the end effector is still on. idrc tbh but if anybody does, i guess a better interface would be to return the processed status.